Helper/Discount.php

```diff
@@ -637,6 +647,24 @@ public function isMirasvitAdminQuoteUsingCreditObserver(Observer $observer){
         return false;
     }
 
+    /**
+     * If enabled, gets the Mirasvit Rewards amount used
+     *
+     * @param \Magento\Quote\Model\Quote $quote  The parent quote of this order which contains Rewards points references
+     *
+     * @return float  If enabled, the currency amount used in the order, otherwise 0
+     */
+    public function getMirasvitRewardsAmount($quote)
+    {
+        /** @var \Mirasvit\Rewards\Helper\Purchase $mirasvitRewardsPurchaseHelper */
+        $mirasvitRewardsPurchaseHelper = $this->mirasvitRewardsPurchaseHelper->getInstance();
+
+        if (!$mirasvitRewardsPurchaseHelper) { return 0; }
+
+        $miravitRewardsPurchase = $mirasvitRewardsPurchaseHelper->getByQuote($quote);
```

Comment:
@mccottry Can reward points be used to pay for shipping and tax? If yes, than fetching the amount assigned to quote before shipping and tax are calculated won't cover the case when the points available to customer are greater than the cart items subtotal. In that case you should find another resource to fetch total customer reward points balance and add it to the discounts (as fixed_amount as you already do). On the other hand, what I wrote may be redundant, it depends on what $mirasvitRewardsPurchaseHelper->getByQuote($quote); does. Perhaps exactly what I pointed above.

Reply:
> Can reward points be used to pay for shipping and tax?

Reply:
> Can reward points be used to pay for shipping and tax?

@danac-gs and @YevhenShBolt  
This question was answered in the PR description's second sentence.
> This PR covers rewards points that include shipping and tax in addition to the cart items.

> On the other hand, what I wrote may be redundant, it depends on what $mirasvitRewardsPurchaseHelper->getByQuote($quote); does.
@danac-gs 
What is written is only redundant because this shipping and tax is usually a first focus for us when implementing store-credit, rewards points, and similar things.  :) Most of us have done several but new guys would need this pointed out to them.  Thanks for being thorough.

Explanation:  
It is not what `$mirasvitRewardsPurchaseHelper->getByQuote($quote);` does, per say, but rather what the interface provides.
The interface for for adding rewards points is on the checkout page along with the interface for estimating shipping and tax.  If one wishes to use the points on shipping and/or tax, they need only enter their address information and get a shipping an tax estimate.  

![image](https://user-images.githubusercontent.com/973856/66800981-af8c3b00-ef17-11e9-9ce2-dfd1fa0599f8.png)